### PR TITLE
linker: merge debug annotations from category c)

### PR DESCRIPTION
**Pull request by pierremoreau**
Fixes: https://github.com/KhronosGroup/SPIRV-Tools/issues/1218

### CODE REVIEW

**Comment by antiagainst** [comment]
Hi @pierremoreau, we have fixed the failure via #1319. You may want to rebase to trigger new builds.

**Comment by pierremoreau** [comment]
Thanks for the notice @antiagainst ; let’s see if it succeeds now.

File: source/link/linker.cpp [hunk 1]
```diff
@@ -445,6 +445,24 @@ static spv_result_t MergeModules(const MessageConsumer& consumer,
       linked_module->AddDebug2Inst(
           std::unique_ptr<Instruction>(inst.Clone(linked_context)));
 
+  for (const auto& module : input_modules)
+    for (const auto& inst : module->debugs3())
+      linked_module->AddDebug3Inst(
+          std::unique_ptr<Instruction>(inst.Clone(linked_context)));
+
+  // If the generated module uses SPIR-V 1.1 or higher, add an
```
Comment (dneto0) [inline source/link/linker.cpp]: This is a nice touch.  :-)

**Comment by dneto0** [comment]
Rebased, test fixed, and pushed into master as 9394272c

File: source/link/linker.cpp (hunk 1 above)
Comment (pierremoreau) [reply source/link/linker.cpp]: :-) I didn’t have it at first, but then decided I might as well make use of it.

**Comment by pierremoreau** [comment]
Thanks for the review and fixing the partial-linkage test.

**Pull request closed without merging**